File: ios/Classes/AppVersionDetailsPlugin.swift
import Flutter
import UIKit

public class AppVersionDetailsPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "app_version_details", binaryMessenger: registrar.messenger())
    let instance = AppVersionDetailsPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getAppVersion":
        if let infoDictionary = Bundle.main.infoDictionary,
           let versionName = infoDictionary["CFBundleShortVersionString"] as? String,
           let versionCode = infoDictionary["CFBundleVersion"] as? String {
            let versionInfo = "\(versionName)+\(versionCode)"
            result(versionInfo)
        } else {
            result(FlutterError(code: "ERROR", message: "Failed to get version info", details: nil))
        }
    case "getPackageName":
        let packageName = Bundle.main.bundleIdentifier ?? "Unknown"
        result(packageName)
    default:
        result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/dev/sivaprasadnk/app_version_details/AppVersionDetailsPlugin.kt
package dev.sivaprasadnk.app_version_details

import android.content.pm.PackageInfo
import android.content.pm.PackageManager
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** AppVersionDetailsPlugin */
class AppVersionDetailsPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will handle communication between Flutter and native Android
  private lateinit var channel: MethodChannel
  private lateinit var packageManager: PackageManager
  private lateinit var packageName: String

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "app_version_details")
    channel.setMethodCallHandler(this)

    // Initialize packageManager and packageName
    val context = flutterPluginBinding.applicationContext
    packageManager = context.packageManager
    packageName = context.packageName
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method == "getAppVersion") {
      try {
        val packageInfo: PackageInfo = packageManager.getPackageInfo(packageName, 0)
        val versionName = packageInfo.versionName
        val versionCode = packageInfo.longVersionCode
        val versionInfo = "$versionName+$versionCode"
        result.success(versionInfo)
      } catch (e: PackageManager.NameNotFoundException) {
        result.error("ERROR", "Failed to get version info", null)
      }
    } else if(call.method =="getPackageName") {
      result.success(packageName);
    }else{
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
